Check agent liveness while migrating

matelakat opened the pull request:
This change makes sure that agent liveness is checked before migrating a router in `l3-agent-migrate` mode.

Please note that the first commit is part of another PR (#30) as well (a small cleanup commit)

Commit:
Testing and code formatting cleanup

Given that now we have multiple scripts, it makes sense to introduce a
test runner. nosetests seems to support our weird file namings, so
using that. It picked up signal_tester as a testcase, so that has been
renamed. Also amended travis, and added noqa for some flake8
violations.

(cherry picked from commit b1d6f841861fe3a1c943de0651b9175926e2049d)

Commit:
l3_agent_check: Do not print target agents

Checking the agent liveness and migrating the agents are two separate
calls of neutron-ha-tool. It does not make sense to print out what
would be the target agent on checking, because:
 - the choice is random
 - the router distribution might change between two calls
Therefore removing the misleading statements, and just listing routers
to be migrated.

Commit:
Test migrate_l3_routers_from_agent

Add some unit tests to cover existing functionality of that method.

Commit:
Refactor - add MigrationResult

`migrate_router_safely` method returned a boolean that indicated the
success of the migration. Given that we want to put a check on agent
liveness to the method, it will be possible that the migration will be
skipped if the agent is alive, thus the return value needs to represent
that fact as well. By making the method return an object, multiple
return values can be added.

Commit:
Check agent liveness during migration

When migrating routers away, make sure that the migration is skipped if
the agent is already back on it's feet.

Commit:
Skip migrating routers after the first skip

When we skipped migrating a router, that means that the agent has came
back alive, so there is no need to call the migration for the rest of
the routers on that specific agent.

vuntz commented on files/default/neutron-ha-tool.py:
```diff
@@ -635,33 +630,43 @@ def migrate_l3_routers_from_agent(qclient, agent, targets, agent_picker,
             if fail_fast:
                 break
+        elif migration_result.skipped:
+            return (migrations, errors)
```
Does it make sense to log something here?

vuntz reviewed (approved):
Generally +1, just a question.

Commit:
Add some log messages

This makes it easier to understand what the tool did, especially in the
case when the agent became online in the meanwhile.

matelakat commented:
Re-basing and adding the change...

matelakat commented:
@vuntz Added a separate commit for adding the log messages.

rhafer reviewed (approved):
lgtm
